packages/ember-metal/lib/property_set.js

```diff
@@ -95,9 +95,9 @@ function setPath(root, path, value, tolerant) {
   let parts = path.split('.');
   let keyName = parts.pop();
 
-  assert('Property set failed: You passed an empty path', keyName && keyName.length > 0)
+  assert('Property set failed: You passed an empty path', keyName.trim().length > 0)
 
-  let newPath = parts.length > 0 ? parts.join('.') : keyName;
```

Comment:
since we already checking if it is path here https://github.com/bekzod/ember.js/blob/14243438756b908b4557f9f3e296d3af89bcda81/packages/ember-metal/lib/property_set.js#L45 parts is guaranteed to be more than 0